src/main/java/io/cdap/plugin/gcp/gcs/sink/GCSOutputFormatProvider.java

```diff
@@ -0,0 +1,274 @@
+        return;
+      }
+
+      FileOutputCommitter fileOutputCommitter = (FileOutputCommitter) delegate;
+      Configuration configuration = taskAttemptContext.getConfiguration();
+      Path committedTaskPath = fileOutputCommitter.getCommittedTaskPath(taskAttemptContext);
+
+      if (configuration == null || committedTaskPath == null) {
+        return;
+      }
+
+      try (FileSystem fileSystem = committedTaskPath.getFileSystem(configuration)) {
+        if (!(fileSystem instanceof GoogleHadoopFileSystem)) {
+          return;
+        }
+        updateInGCSFS(taskAttemptContext, configuration, committedTaskPath,
+                      (GoogleHadoopFileSystem) fileSystem);
+
+      }
+    }
+
+    private void updateInGCSFS(TaskAttemptContext taskAttemptContext, Configuration configuration,
+                               Path committedTaskPath, GoogleHadoopFileSystem fileSystem) throws IOException {
+      GoogleCloudStorageFileSystem gcsFs = fileSystem.getGcsFs();
+      if (gcsFs == null) {
+        return;
+      }
+
+      try {
+        List<URI> files = gcsFs.listFileNames(gcsFs.getFileInfo(committedTaskPath.toUri()));
+        if (files == null || files.isEmpty()) {
+          return;
+        }
+        //update metadata on any file - typically only one will be there
+        GCSPath objPath = GCSPath.from(files.get(0).toString());
+        //read the count from configuration
+        String keyInConfig = String.format(RECORD_COUNT_FORMAT, taskAttemptContext.getTaskAttemptID());
+        Map<String, byte[]> metaData = new HashMap<>();
+        byte[] bytes = Long.valueOf(configuration.getLong(keyInConfig, 0)).toString().getBytes();
+        metaData.put(GCSBatchSink.RECORD_COUNT, bytes);
+        UpdatableItemInfo info = new UpdatableItemInfo(
+          new StorageResourceId(objPath.getBucket(), objPath.getName()), metaData);
+        gcsFs.getGcs().updateItems(Collections.singletonList(info));
+      } finally {
+        gcsFs.close();
+      }
+    }
+
+    @Override
+    public void abortTask(TaskAttemptContext taskAttemptContext) throws IOException {
+      delegate.abortTask(taskAttemptContext);
+    }
+
+    @Override
+    public boolean isCommitJobRepeatable(JobContext jobContext) throws IOException {
+      return delegate.isCommitJobRepeatable(jobContext);
+    }
+
+    @Override
+    public boolean isRecoverySupported(JobContext jobContext) throws IOException {
+      return delegate.isRecoverySupported(jobContext);
+    }
+
+    @Override
+    public boolean isRecoverySupported() {
+      return delegate.isRecoverySupported();
+    }
+
+    @Override
+    public void recoverTask(TaskAttemptContext taskContext) throws IOException {
+      delegate.recoverTask(taskContext);
+    }
+  }
+
+  /**
+   * RecordWriter for GCSSink
+   */
+  public static class GCSRecordWriter extends RecordWriter<NullWritable, StructuredRecord> {
+
+    private RecordWriter originalWriter;
+    private AtomicLong recordCount;
```

Comment:
don't think this needs to be an AtomicLong, write() isn't called concurrently

Reply:
Removed.